.github/workflows/main.yml

```diff
@@ -0,0 +1,127 @@
+name: CI/CD
+on: [push, pull_request]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      
+      - name: Setup node
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
+
+      - name: Get yarn cache
+        id: yarn-cache
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - uses: actions/cache@v1
+        with:
+          path: ${{ steps.yarn-cache.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+      
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Build project
+        run: yarn build
+
+      - name: Upload build artifact
+        uses: actions/upload-artifact@v1
```

Comment:
What does this do?

Reply:
We save build artifacts (which is `lib` directory) to the cloud to re-use them in other jobs.
So we build **only once**, and then we test artifacts on different environments and then release these tested artifacts to the npm.

Reply:
Oh nice!

Reply:
BTW - I didn't notice a build being pushed out after you merged https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/pull/365

Is our semantic release still working?

Reply:
It should work. The beta branch was only 3 _chore_ commits ahead so semantic-release didn’t run real release, only did dry run to ensure if setup is correct (it really test it because I had to set new tokens for it to work)

Reply:
Cool. It would be good to test it for real as you're making changes so we know it's all still good. I've kind of given up on npm giving me publishing rights on fork-ts-checker-webpack-plugin. Despite all your efforts as well as my own it still hasn't happened. It's super frustrating but as long as we've got an automated release process that works we can cope. Hence I'm slightly paranoid about it 😄

Reply:
Yes, the npm tech support wasn't very helpful 🙁 We will test semantic-release after merging #364 into the beta branch 🙂

Reply:
Excellent!